### PR TITLE
Dialects: finishing Arith custom parse and printing

### PR DESCRIPTION
**Pull request by ajrichins**
adding simplified form parsing and printing for arith operations select, cmpi, cmpf, truncf, negf and extf.

added custom parsing filecheck test: @superlopuh 

Ignore commits previous to the merge, there is no changes there that aren't already in main.  I'll make sure to start fresh next time.

### CODE REVIEW

File: xdsl/dialects/arith.py [hunk 1]
```diff
@@ -418,6 +418,43 @@ def __init__(
...
+        input_type = parser.parse_type()
+        (operand1, operand2) = parser.resolve_operands(
+            [operand1, operand2], 2 * [input_type], parser.pos
+        )
+
+        return cls(operand1, operand2, arg)
+
+    def print(self, printer: Printer):
+        printer.print(" ")
+        cmpi_comparison_operations = [
```
Comment (superlopuh) [inline xdsl/dialects/arith.py:438]: Can you please move this to `_CMPI_COMPARISON_OPERATIONS` either on the file or module level?

File: xdsl/dialects/arith.py [hunk 2]
```diff
@@ -489,6 +526,49 @@ def __init__(
...
```
Comment (superlopuh) [inline xdsl/dialects/arith.py:546]: same here

**Review by superlopuh — CHANGES_REQUESTED** [review]
thank you! jsut a minor change would be good

**Comment by superlopuh** [comment]
Looks like you might need to re-run the notebooks, some print MLIR, and the CI tests for the resulting text matching a re-run exactly.

**Comment by ajrichins** [comment]
yeah, i had to run to class right after submitting :( I'll get those done rn

File: xdsl/dialects/arith.py (hunk 1 above)
Comment (ajrichins) [reply xdsl/dialects/arith.py]: put it by other file globals

File: xdsl/dialects/arith.py (hunk 2 above)
Comment (ajrichins) [reply xdsl/dialects/arith.py]: put it by other file globals

File: docs/database_example.ipynb [hunk 3]
```diff
@@ -471,8 +471,16 @@
    "name": "python3"
   },
   "language_info": {
+   "codemirror_mode": {
```
Comment (superlopuh) [inline docs/database_example.ipynb:474]: this is a bit tedious, but could you please modify this notebook by hand to remove this diff?

File: docs/xdsl-introduction.ipynb [hunk 4]
```diff
@@ -843,8 +843,22 @@
   }
  ],
  "metadata": {
+  "kernelspec": {
```
[the remark this answers is not in the document]
Comment (ajrichins) [reply docs/xdsl-introduction.ipynb:846]: should be fixed

File: docs/database_example.ipynb (hunk 3 above)
Comment (ajrichins) [reply docs/database_example.ipynb]: should be fixed

**Review by superlopuh — APPROVED** [review]
ty

**Comment by superlopuh** [comment]
One thing that's missing is a filecheck for custom syntax MLIR interop. Can you please do something similar to what I do in this PR, which is to duplicate the generic syntax filecheck file, and convert it to custom syntax instead? #1545 Annoyingly, we can't parse MLIR custom syntax directly since MLIR prints `module` and we need `builtin.module`...

**Comment by ajrichins** [comment]
just noticed there is only 3 ops without the custom syntax, truncf, extf and negf so I will try and add those before finishing the PR.

**Comment by ajrichins** [comment]
Can't figure out why the 3.10 build failed the MLIR testing other than it seems to be a network issue?

**Comment by tobiasgrosser** [comment]
This looks all great. Thank you @ajrichins for your contribution.

**Comment by superlopuh** [comment]
@ajrichins I saw that you added a test with XDSL_ROUNDRIP, I'm not sure whether that was to address my comment. Is the MLIR interop test still upcoming?

File: xdsl/dialects/arith.py [hunk 5]
```diff
@@ -42,6 +42,38 @@
...
+    "ole",
+    "one",
+    "ord",
+    "ueq",
+    "ugt",
+    "uge",
+    "ult",
+    "ule",
+    "une",
+    "uno",
```
Comment (math-fehr) [inline xdsl/dialects/arith.py:73]: Can you make these two constants public (without the `_`)?
In case we ever need to access it from outside.

**Review by math-fehr — APPROVED** [review]
Very nice!
I just added one comment, and then this is good to merge for me!

**Comment by ajrichins** [comment]
> 

ahh I did misunderstand, I'll get that added

**Pull request merged**